libsolutil/JSON.cpp

```diff
@@ -33,8 +33,8 @@
 using namespace std;
 
 static_assert(
-	(JSONCPP_VERSION_MAJOR == 1) && (JSONCPP_VERSION_MINOR == 9) && (JSONCPP_VERSION_PATCH == 3),
-	"Unexpected jsoncpp version: " JSONCPP_VERSION_STRING ". Expecting 1.9.3."
+	(NLOHMANN_JSON_VERSION_MAJOR == 3) && (NLOHMANN_JSON_VERSION_MINOR == 10) && (NLOHMANN_JSON_VERSION_PATCH == 2),
+	"Unexpected nlohmann-json version. Expecting 3.10.2."
 );
```

Comment:
Do we actually need to enforce one specific version? Why not allow a range?

Reply:
Not sure, just kept the same logic.

Reply:
Well we download a specific file too, so this is just a safety check. I think it is fine.